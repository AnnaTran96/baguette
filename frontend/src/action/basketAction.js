import { ADD_TO_BASKET, ADD_QUANTITY, SUBTRACT_QUANTITY, REMOVE_ITEM, ADD_SHIPPING, SUBTRACT_SHIPPING } from './actions'

export const addToBasket = id => {
    return {
        type: ADD_TO_BASKET, id
    }
}

export const addQuantity = id => {
    return {
        type: ADD_QUANTITY, id
    }
}

export const subtractQuantity = id => {
    return {
        type: SUBTRACT_QUANTITY, id
    }
}

export const removeItem = id => {
    return {
        type: REMOVE_ITEM, id
    }
}

export const addShipping = () => {
    return {
        type: ADD_SHIPPING
    }
}

export const subtractShipping = () => {
    return {
        type: SUBTRACT_SHIPPING
    }
}