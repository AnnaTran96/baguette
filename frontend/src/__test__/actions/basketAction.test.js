import * as actions from '../../action/basketAction'
import configureStore from 'redux-mock-store'

const mockStore = configureStore()
const store = mockStore()

describe('Actions', () => {
    
    beforeEach(() => {
        store.clearActions()
    })

    it('should create an action to add to the basket', () => {
        const expectedAction = {
            type: "ADD_TO_BASKET",
            
        }
        expect(store.getActions()).toMatchSnapshot()
    })

    it('should increase quantity of item', () => {
        const expectedAction = {
            type: "ADD_QUANTITY",
            
        }
        expect(store.getActions()).toMatchSnapshot()
    })
})