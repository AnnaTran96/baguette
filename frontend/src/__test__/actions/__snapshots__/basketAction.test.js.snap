// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Actions should create an action to add to the basket 1`] = `Array []`;

exports[`Actions should increase quantity of item 1`] = `Array []`;
